Added Viewport Texture Tutorial

clayjohn opened the pull request:
This tutorial explains how to use a Viewport to generate a texture that can be applied to a sphere. In order to do so it walks the reader through the process of making a basic procedural planet.

In the future I would like to follow this tutorial up with two others. One showing how to create animated normal maps with shaders. And another showing how to use shaders to create an atmosphere effect for planets.

Please let me know if anything can be improved! I am very open to comments and constructive criticism.

YeldhamDev commented on tutorials/viewports/using_viewport_as_texture.rst:
```diff
@@ -0,0 +1,321 @@
+.. _doc_viewport_as_texture:
+
+Using a Viewport as a texture
+=====================================
```
You might want to remove the extraneous `=`'s.

clayjohn commented:
Good catch. Thanks!

cbscribe commented on tutorials/viewports/using_viewport_as_texture.rst:
```diff
@@ -0,0 +1,321 @@
+
+.. image:: img/planet_example.png
+
+.. note:: This tutorial does not cover how to code a dynamic atmosphere like the one this planet has
```
Missing a . at the end of the sentence.

cbscribe commented on tutorials/viewports/using_viewport_as_texture.rst:
```diff
@@ -0,0 +1,321 @@
+
+.. image:: img/planet_seam.png
+
+Notice the ugly seem that forms where the texture texture wraps around? This is because we are picking 
```
seem -> seam

cbscribe commented on tutorials/viewports/using_viewport_as_texture.rst:
```diff
@@ -0,0 +1,321 @@
+3D noise to determine a color value. By calculating the noise in 3D we solve the problem
+of the pinching at the poles. To understand why, picture the noise being calculated across the 
+surface of the sphere instead of across the 2D plane. When you calculate across the 
+surface of the sphere you never hit an edge, and hence you never create a seem or
```
seem -> seam

cbscribe commented on tutorials/viewports/using_viewport_as_texture.rst:
```diff
@@ -0,0 +1,321 @@
+
+.. image:: img/planet_noise.png
+
+.. note:: In order to highlight the texture, I have turned set the material to unshaded.
```
remove "turned"

cbscribe commented on tutorials/viewports/using_viewport_as_texture.rst:
```diff
@@ -0,0 +1,321 @@
+
+One more thing to make this a little more planet-y. The land shouldn't be so blobby lets make the edges
+a little rougher. A trick that is often used in shaders to make rough looking terrain with noise is
+to layer levels of noise over one another and various frequencies. We use one layer to make the 
```
I think "and" here should be a different word. "levels of noise over one another *using* various frequencies". "at" could work there as well.

cbscribe commented on tutorials/viewports/using_viewport_as_texture.rst:
```diff
@@ -0,0 +1,321 @@
+a little rougher. A trick that is often used in shaders to make rough looking terrain with noise is
+to layer levels of noise over one another and various frequencies. We use one layer to make the 
+overall blobby structure of the continents. Then another layer breaks up the edges a bit, and then
+another and so on. What we will do is calculate ``n`` with four lines of shader code
```
", and so on."

clayjohn commented:
Thanks @cbscribe! I addressed all your comments. 

I still can't believe I misspelled "seam" so many times. I guess thats what I get for writing this in vim...

cbscribe commented:
I gave it some time in case others wanted to take a look. This is good content, so let's merge it!
